Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/storage/memory/MemoryNodeTest.java

```diff
@@ -0,0 +1,947 @@
+
+        MemoryNode condition = new MemoryNode(Mods.NAME).put(RDF.toURL(9),
+            new MemoryNode("http://names.example/alice"));
+
+        assertThat(node.matches(condition), is(false));
+    }
+
+    /**
+     * Tests {@code MemoryNode()} constructor.
+     */
+    @Test
+    public void testMemoryNodeCanBeCreatedEmpty() {
+        new MemoryNode();
+    }
+
+    /**
+     * Tests {@code MemoryNode(IdentifiableNode)} constructor.
+     */
+    @Test
+    public void testMemoryNodeCanBeCreatedWithMemoryNodeReferenceAsType() {
+        MemoryNode mets = new MemoryNode(Mets.METS);
+
+        assertThat(mets.getType(), is(equalTo(Mets.METS.getIdentifier())));
+    }
+
+    /**
+     * Tests {@code MemoryNode(String)} constructor.
+     */
+    @Test
+    public void testMemoryNodeCanBeCreatedWithStringAsType() {
+        final String MODS = Mods.MODS.getIdentifier();
+        MemoryNode mods = new MemoryNode(MODS);
+
+        assertThat(mods.getType(), is(equalTo(MODS)));
+    }
+
+    /**
+     * Tests {@code MemoryNode(String)} constructor.
+     */
+    @Test(expected = AssertionError.class)
+    public void testMemoryNodeCreatedWithEmptyStringAsTypeIsEmpty() {
+        try {
+            assert (false);
+            fail("Assertions disabled. Run JVM with -ea option.");
+        } catch (AssertionError e) {
+            new MemoryNode("");
+        }
+    }
+
+    /**
+     * Tests {@code MemoryNode(IdentifiableNode)} constructor.
+     */
+    @Test
+    public void testMemoryNodeCreatedWithUnitializedMemoryNodeReferenceIsEmpty() {
+        MemoryNode a = new MemoryNode((MemoryNodeReference) null);
+        assertThat(a.isEmpty(), is(true));
+    }
+
+    /**
+     * Tests {@code MemoryNode(String)} constructor.
+     */
+    @Test
+    public void testMemoryNodeCreatedWithUnitializedStringIsEmpty() {
+        MemoryNode a = new MemoryNode((String) null);
+        assertThat(a.isEmpty(), is(true));
+    }
+
+    /**
+     * Tests {@code putAll(String, Collection<? extends ObjectType>)}.
+     */
+    @Test
+    public void testPutAll() {
+        List<ObjectType> nodes = new ArrayList<>();
+        nodes.add(new MemoryNodeReference("http://names.example/alice"));
+        nodes.add(new MemoryNodeReference("http://names.example/bob"));
+        nodes.add(new MemoryNodeReference("http://names.example/charlie"));
+
+        Node node = new MemoryNode();
+        node.putAll(Mods.NAME.getIdentifier(), nodes);
+
+        assertThat(node.get(Mods.NAME).countUntil(Long.MAX_VALUE), is((long) 3));
+    }
+
+    /**
+     * Tests {@code put(String, ObjectType)}.
+     */
+
+    @Test
+    public void testPutStringObjectType() {
+        new MemoryNode().put("http://www.loc.gov/mods/v3#name", MemoryLiteral.createLeaf("Wilhelm Busch", "de"));
+    }
+
+    /**
+     * Tests {@code put(String, String)}.
+     */
+    @Test
+    public void testPutStringString() {
+        new MemoryNode().put("http://www.loc.gov/mods/v3#name", "Wilhelm Busch");
```

Comment:
What are you testing here?

Reply:
Test that put method which takes two string arguments works. (no exception being thrown)